packages/nexrender-core/src/tasks/download.js

```diff
@@ -37,15 +38,8 @@ const download = (job, settings, asset) => {
 
     /* try to guess the extension from data part */
     if (protocol == 'data' && !asset.extension) {
-        let databuf = data2buf(asset.src)
-
-        switch (databuf.type) {
-            case 'image/png': asset.extension = 'png'; break;
-            case 'image/gif': asset.extension = 'gif'; break;
-            case 'image/bmp': asset.extension = 'bmp'; break;
-            case 'image/jpeg': asset.extension = 'jpg'; break;
-            case 'application/json': asset.extension = 'json'; break;
-        }
+        const databuf = data2buf(asset.src)
+        asset.extension = mime.extension(databuf.type) || ''
```

Comment:
You should not update extension to `''` if it can not be resolved, but rather leave it `undefined`..